homeassistant/components/proxmoxve/__init__.py

```diff
@@ -91,181 +107,387 @@
+    proxmox_client = ProxmoxClient(host, port, user, realm, password, verify_ssl)
+    try:
+        await hass.async_add_executor_job(proxmox_client.build_client)
+    except AuthenticationError as error:
+        raise ConfigEntryAuthFailed from error
+    except SSLError as error:
+        raise ConfigEntryNotReady(
+            f"Unable to verify proxmox server SSL. Try using 'verify_ssl: false' for proxmox instance {host}:{port}"
+        ) from error
+    except ConnectTimeout as error:
+        raise ConfigEntryNotReady(
+            f"Connection to host {host} timed out during setup"
+        ) from error
+    except ResourceException as error:
+        raise ConfigEntryNotReady from error
+
+    proxmox = await hass.async_add_executor_job(proxmox_client.get_api_client)
+
+    async def async_update(
+        api_category: str,
+        node: str | None = None,
+        vm_id: int | None = None,
+    ) -> dict:
+        """Update the API data."""
 
-        for node_config in host_config["nodes"]:
-            node_name = node_config["node"]
-            node_coordinators = coordinators[host_name][node_name] = {}
+        def poll_api():
+            """Call the api."""
+            api_status = {}
 
-            for vm_id in node_config["vms"]:
-                coordinator = create_coordinator_container_vm(
-                    hass, proxmox, host_name, node_name, vm_id, TYPE_VM
+            try:
+                if api_category == ProxmoxType.Proxmox:
+                    api_status = proxmox.version.get()
+                elif api_category is ProxmoxType.Node:
+                    api_status = get_data_node(proxmox, node)
+                elif api_category == ProxmoxType.QEMU:
+                    api_status = proxmox.nodes(node).qemu(vm_id).status.current.get()
+                elif api_category == ProxmoxType.LXC:
+                    api_status = proxmox.nodes(node).lxc(vm_id).status.current.get()
+            except AuthenticationError as error:
+                raise ConfigEntryAuthFailed from error
+            except SSLError as error:
+                raise UpdateFailed from error
+            except ConnectTimeout as error:
+                raise UpdateFailed from error
+            except ResourceException as error:
+                raise UpdateFailed from error
+
+            return api_status
+
+        try:
+            status = await hass.async_add_executor_job(poll_api)
+        except ResourceException as error:
+            raise UpdateFailed(error) from error
+
+        return parse_api_proxmox(status, api_category)
+
+    async def async_init_coordinator(
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize a RainMachineDataUpdateCoordinator."""
+        await coordinator.async_config_entry_first_refresh()
+
+    controller_init_tasks = []
+    coordinators = {}
+    for api_category, update_interval in COORDINATOR_UPDATE_INTERVAL_MAP.items():
```

Comment:
Hey @bachya,

Is it acceptable to customize the coordinators update interval this way?

```
    coordinator_interval_update_map: dict[ProxmoxType, timedelta] = {
        ProxmoxType.Proxmox: timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL_HOST]),
        ProxmoxType.Node: timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL_NODE]),
        ProxmoxType.QEMU: timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL_QEMU]),
        ProxmoxType.LXC: timedelta(seconds=config_entry.data[CONF_SCAN_INTERVAL_LXC]),
    }
```

The configuration entry would be created with the default values, but could be changed in the options flow. I tested it here and it works fine, but I want to know if it's acceptable to do this in core before submitting the change.

Thanks!

Reply:
Yes, that's fine – however...

> could be changed in the options flow

We generally don't allow users to alter update intervals (since it often leads to confusion). You should pick sensible defaults on behalf of the user.

Reply:
It's just that this sensible time is relative according to what each user expects from the integration and how many VM/CT he adds to it, because each VM/CT will be a request to the server.
Currently the time is 60 seconds, but only with a binary sensor implemented.
With the implementation of sensors this time may seem too long for some users. Reducing this time can make more conservative users annoyed by the amount of requests to the server.

That's why I thought about this implementation, which when well documented I believe it will contribute. It would be the last step of the options flow.

![image](https://user-images.githubusercontent.com/31328123/190001694-bff20608-219d-4a76-adc5-12d8b973dd56.png)

If this implementation is not indicated, I think about using 30 seconds to update the VM/CT coordinator.

Reply:
You can certainly give it a shot; I've just experienced with my own PRs that user adjustment of update intervals hasn't been approved.